### PR TITLE
nixos/generic-extlinux-compatible: add mirroredBoots option

### PR DESCRIPTION
**Pull request by Luflosi**
## Description of changes
Analogous to `boot.loader.grub.mirroredBoots`.
This allows installing the configuration files to multiple locations.

I have a Raspberry Pi 4 with U-Boot installed on a microSD card and the root filesystem (ZFS) mirrored on two USB disks. /boot was so far only stored on one of the disks. This frequently caused the system to not boot because the two disks would show up to U-Boot in a random order. This sometimes caused U-Boot to try to boot off of the wrong disk, which does not have the /boot partition. I'm not aware of any way to tell U-Boot which of the two USB disks is the correct one. Storing a boot partition on every disk fixes this problem.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Luflosi** [comment]
@Lassulus could you take a look please?

**Review by Lassulus — APPROVED** [review]
lgtm, although I have no prior experience with mirroredBoots.

**Review by Mic92 — APPROVED** [review]
Looks good to me in principal. I have to wait a few days until I can test this on an actual system. If someone else can test it, go ahead. I am bit more cautious with bootloader changes even if they are simple.

File: nixos/modules/system/boot/loader/generic-extlinux-compatible/default.nix [hunk 1]
```diff
@@ -72,11 +96,25 @@ in
...
+      system.build.installBootLoader = installBootLoader;
       system.boot.loader.id = "generic-extlinux-compatible";
 
       boot.loader.generic-extlinux-compatible.populateCmd = "${populateBuilder} ${builderArgs}";
+
+      assertions = [
+        {
+          assertion = cfg.mirroredBoots != [ ];
+          message = "You must not remove all elements from option "
+            + "'boot.loader.generic-extlinux-compatible.mirroredBoots', otherwise the system will not be bootable.";
```
[the remark this answers is not in the document]
Comment (Luflosi) [reply nixos/modules/system/boot/loader/generic-extlinux-compatible/default.nix]: Thanks for this suggestion!

**Comment by Luflosi** [comment]
@Mic92 did you have some time to test it yet?

**Comment by Luflosi** [comment]
Thank you!

**Pull request merged**